api/include/opentelemetry/logs/logger.h

```diff
@@ -38,8 +41,78 @@ class Logger
   virtual const nostd::string_view GetName() noexcept = 0;
 
   /**
-   * Each of the following overloaded Log(...) methods
-   * creates a log message with the specific parameters passed.
+   * Create a Log Record object
+   *
+   * @return nostd::unique_ptr<LogRecord>
+   */
+  virtual nostd::unique_ptr<LogRecord> CreateLogRecord() noexcept = 0;
+
+  /**
+   * Emit a Log Record object with event
+   *
+   * @param event_name event.name
+   * @param log_record
+   */
+  virtual void EmitEvent(nostd::string_view event_name,
```

Comment:
Not for immediate attention, but just to update that Log/Event API specs has changed again :( - https://github.com/open-telemetry/opentelemetry-specification/pull/2941/

We can accommodate these changes separately in another PR. Will create a task for that.

Reply:
OK, I removed `EmitEvent` and will add it back into `EventLogger` later when implementing #1687 . We can just modify #1687 to follow the changes.